src/dryad/LegendreAngularDistribution/test/LegendreAngularDistribution.test.cpp

```diff
@@ -0,0 +1,350 @@
+        CHECK_THAT(  0.625    , WithinRel( linear.cosines()[10] ) );
+        CHECK_THAT(  0.6875   , WithinRel( linear.cosines()[11] ) );
+        CHECK_THAT(  0.75     , WithinRel( linear.cosines()[12] ) );
+        CHECK_THAT(  0.8125   , WithinRel( linear.cosines()[13] ) );
+        CHECK_THAT(  0.875    , WithinRel( linear.cosines()[14] ) );
+        CHECK_THAT(  0.90625  , WithinRel( linear.cosines()[15] ) );
+        CHECK_THAT(  0.9375   , WithinRel( linear.cosines()[16] ) );
+        CHECK_THAT(  0.96875  , WithinRel( linear.cosines()[17] ) );
+        CHECK_THAT(  0.984375 , WithinRel( linear.cosines()[18] ) );
+        CHECK_THAT(  0.9921875, WithinRel( linear.cosines()[19] ) );
+        CHECK_THAT(  1.0      , WithinRel( linear.cosines()[20] ) );
+
+        CHECK_THAT( -30.0             , WithinRel( linear.values()[0] ) );
+        CHECK_THAT( -22.859375        , WithinRel( linear.values()[1] ) );
+        CHECK_THAT( -16.875           , WithinRel( linear.values()[2] ) );
+        CHECK_THAT( -11.953125        , WithinRel( linear.values()[3] ) );
+        CHECK_THAT( -9.861328125      , WithinRel( linear.values()[4] ) );
+        CHECK_THAT( -8.0              , WithinRel( linear.values()[5] ) );
+        CHECK_THAT( -6.357421875      , WithinRel( linear.values()[6] ) );
+        CHECK_THAT( -4.921875         , WithinRel( linear.values()[7] ) );
+        CHECK_THAT( -3.681640625      , WithinRel( linear.values()[8] ) );
+        CHECK_THAT( -2.625            , WithinRel( linear.values()[9] ) );
+        CHECK_THAT( -1.740234375      , WithinRel( linear.values()[10] ) );
+        CHECK_THAT( -1.358642578125   , WithinRel( linear.values()[11] ) );
+        CHECK_THAT( -1.015625         , WithinRel( linear.values()[12] ) );
+        CHECK_THAT( -0.709716796875   , WithinRel( linear.values()[13] ) );
+        CHECK_THAT( -0.439453125      , WithinRel( linear.values()[14] ) );
+        CHECK_THAT( -0.317230224609375, WithinRel( linear.values()[15] ) );
+        CHECK_THAT( -0.203369140625   , WithinRel( linear.values()[16] ) );
+        CHECK_THAT( -0.097686767578125, WithinRel( linear.values()[17] ) );
+        CHECK_THAT( -0.047855377197266, WithinRel( linear.values()[18], 1e-12 ) );
+        CHECK_THAT( -0.023682117462158, WithinRel( linear.values()[19], 1e-12 ) );
+        CHECK_THAT(  0.0              , WithinAbs( linear.values()[20], 1e-12 ) );
+
+        CHECK( true == linear.isLinearised() );
+      } // THEN
+
+      THEN( "arithmetic operations can be performed" ) {
+
+        LegendreAngularDistribution small( { 3., 0., 1. } );
+        LegendreAngularDistribution equal( { 3., 0., 0., 1. } );
+        LegendreAngularDistribution large( { 3., 0., 0., 0., 1. } );
+        LegendreAngularDistribution result( { 0. } );
+
+        result = -chunk;
+
+        CHECK( 3 == result.order() );
+        CHECK( 4 == result.coefficients().size() );
+        CHECK_THAT(  10.33333333333333, WithinRel( result.coefficients()[0] ) );
```

Comment:
I would almost suggest testing these in linearized space where assumptions can be verified easier, but then it would depend on the implementation detail of linearization. But all of these look correct.